llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -21345,18 +21345,17 @@ bool RISCVTargetLowering::splitValueIntoRegisterParts(
   }
 
   if (ValueVT.isRISCVVectorTuple() && PartVT.isRISCVVectorTuple()) {
+#ifndef NDEBUG
```

Comment:
Is there ever case where PartVT isn't exactly equal to ValueVT? Why do we need all the math?

Reply:
It's because of fractional LMUL, I think LMUL = 1 and fractional LMUL use the same register class, do we still need a bitcast?

Reply:
If the types don't match then I think we need some operation to represent the type change. So I guess the original code is correct.

Reply:
I see, let me close this PR.